extensions/lifecycle/conductor/LifecycleConductor.js

```diff
@@ -198,50 +293,27 @@ class LifecycleConductor {
             // already initialized
             return process.nextTick(done);
         }
+
+        this._setupVaultClientCache();
         return async.series([
-            next => {
-                const producer = new BackbeatProducer({
-                    kafka: { hosts: this.kafkaConfig.hosts },
-                    topic: this.lcConfig.bucketTasksTopic,
-                });
-                producer.once('error', next);
-                producer.once('ready', () => {
-                    this.logger.debug(
-                        'producer is ready',
-                        { kafkaConfig: this.kafkaConfig,
-                          topic: this.lcConfig.bucketTasksTopic });
-                    producer.removeAllListeners('error');
-                    producer.on('error', err => {
-                        this.logger.error('error from backbeat producer', {
-                            topic: this.lcConfig.bucketTasksTopic,
-                            error: err,
-                        });
-                    });
-                    this._producer = producer;
-                    next();
-                });
-            },
-            next => {
-                this._zkClient = zookeeperHelper.createClient(
-                    this.zkConfig.connectionString);
-                this._zkClient.connect();
-                this._zkClient.once('error', next);
-                this._zkClient.once('ready', () => {
-                    // just in case there would be more 'error' events
-                    // emitted
-                    this._zkClient.removeAllListeners('error');
-                    this._zkClient.on('error', err => {
-                        this.logger.error(
-                            'error from lifecycle conductor zookeeper client',
-                            { error: err });
-                    });
-                    next();
-                });
-            },
+            next => this._setupProducer(next),
+            next => this._setupZookeeperClient(next),
             next => this._initKafkaBacklogMetrics(next),
         ], done);
     }
 
+    _setupVaultClientCache() {
+        if (this.lcConfig.auth.type !== authTypeAssumeRole) {
+            return;
+        }
+
+        this._vaultClientCache = new VaultClientCache();
+        const { host, port } = this.lcConfig.auth.vault;
+        this._vaultClientCache
```

Comment:
How do we set the credential keys (access key/secret key) to authenticate the Vault client call?

Reply:
this is temporary and will be fixed; the `getAccountIds` api does not require access/secret keys for authentication, and only need to be from an internal ip